x-pack/plugins/spaces/public/ui_api/lazy_wrapper.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC, PropsWithChildren, PropsWithRef, ReactElement } from 'react';
+import React, { lazy, useEffect, useMemo, useState } from 'react';
+
+import type { NotificationsStart, StartServicesAccessor } from 'src/core/public';
+
+import type { PluginsStart } from '../plugin';
+import { SuspenseErrorBoundary } from '../suspense_error_boundary';
+
+interface InternalProps<T> {
+  fn: () => Promise<FC<T>>;
+  getStartServices: StartServicesAccessor<PluginsStart>;
+  props: JSX.IntrinsicAttributes & PropsWithRef<PropsWithChildren<T>>;
+}
+
+export const LazyWrapper: <T>(props: InternalProps<T>) => ReactElement = ({
+  fn,
+  getStartServices,
+  props,
+}) => {
+  const [notifications, setNotifications] = useState<NotificationsStart | undefined>(undefined);
+  useEffect(() => {
+    getStartServices().then(([coreStart]) => {
+      setNotifications(coreStart.notifications);
+    });
+  });
```

Comment:
Again, the side effect here will run at every render unless you add a dependency array and you also need to ensure the component is still mounted when modifying state asynchronously.

What you could do so you don't have to manage all that yourself is use the `useAsync` hook from `react-use` module:

```ts
const { value: { notifications } } = useAsync(getStartServices); // `value` is the `coreStart` object returned by `getStartServices`

return <SuspenseErrorBoundary notifications={notifications} />
```

https://streamich.github.io/react-use/?path=/story/side-effects-useasync--docs

Reply:
Good idea, thanks!
Destructuring makes this a bit funky, but this is what I did to make it work:
```
  const { value: startServices = [{ notifications: undefined }] } = useAsync(getStartServices);
  const [{ notifications }] = startServices;
```